File: State/State/Context.m
//
//  Context.m
//  State
//
//  Created by ZhangYu on 2017/9/4.
//  Copyright © 2017年 DeveloperZhang. All rights reserved.
//

#import "Context.h"
#import "State.h"

@interface Context ()

@end

@implementation Context

- (id)initWithState:(State *)state {
    self = [super init];
    if (self) {
        self.state = state;
    }
    return self;
}

- (void)setState:(State *)state {
    _state = state;
    NSLog(@"state is:%@",NSStringFromClass([self.state class]));
}

- (void)request {
    [self.state handle:self];
}

@end

File: Strategy/Strategy/Context.swift
//
//  Context.swift
//  Strategy
//
//  Created by ZhangYu on 2017/9/12.
//  Copyright © 2017年 DeveloperZhang. All rights reserved.
//

import UIKit

class Context: NSObject {
    var strategy: Strategy;
    
    init(strategy:Strategy) {
        self.strategy = strategy;
    }
    
    func contextInterface() {
        strategy.algorithmInterface();
    }
}
